src/main/java/nextstep/ladder/model/Point.java

```diff
@@ -4,18 +4,17 @@
 
 public class Point {
     private boolean value;
-    private LineGenerator lineGenerator;
 
-    public Point(LineGenerator lineGenerator) {
-        this.lineGenerator = lineGenerator;
+    public Point() {
+        this.value = false;
     }
 
     public Point(boolean value) {
         this.value = value;
     }
 
-    public void next(boolean hasPreviousLine) {
-        this.value = !hasPreviousLine && lineGenerator.generate();
+    public void next(boolean hasPreviousLine, boolean currentLine) {
+        this.value = !hasPreviousLine && currentLine;
```

Comment:
이전 값을 받아 다음 값을 자신의 필드에게 할당하는 것도 좋지만,
Point 객체를 불변객체로 만들어 자신의 값에 따라 다음 값을 생성해서 리턴하는 방법도 고려해 보시면 좋을 것 같아요
```suggestion
    public Point next(...) {
        return new Point(...);
```